lib/retry/WaitStrategy.ts

```diff
@@ -0,0 +1,35 @@
+import { WaitPattern } from './RetryOptions';
+
+export class WaitStrategy {
+
+  constructor(
+    private readonly waitPattern: WaitPattern,
+  ) { }
+
+  public wait(index: number): Promise<void> {
+    if (!this.waitPattern) {
+      return Promise.resolve();
+    }
+
+    const timeout = this.getTimeout(index) || 0;
+    return new Promise(resolve => setTimeout(resolve, timeout));
+  }
+
+  private getTimeout(index: number): number {
+    if (Array.isArray(this.waitPattern)) {
+      const values = this.waitPattern as number[];
+      const count = values.length;
+      return index > count ? values[count - 1] : values[index];
```

Comment:
Shouldn't it be `index >= count`?

Reply:
No, because `count` is length of an array, and if count (`array.length`) will be equal with 0, function will search to return an element from array, and it will throw an error.